### PR TITLE
Connecting to api

### PR DESCRIPTION
**Pull request by Therealadetunji**
Fetch book list from Bookstore API on Books page load.  ✔️ 
Add an Action Creator, that returns a function.  ✔️ 
Fetch list of books.  ✔️ 
Dispatch another action (add an action type) that will pass API response to reducer and updates the state. ✔️ 
Refactor your add book and remove book features to persist your changes in the server. ✔️ 
Change your existing Action Creators, so they return functions. ✔️ 
Make changes by using API. ✔️

### CODE REVIEW

File: src/redux/Books/Books.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        ...state, action.book,
+      ];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.item_id);
+    case getBook:
+      return action.payload;
+    default:
+      return state;
+  }
+};
```
Comment (cwaku) [inline src/redux/Books/Books.js]: - [ ] As per the project requirement, kindly make use of `createAsyncThunk` for creating thunk action creators:

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @Therealadetunji 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️  but you are almost there! 🏃🏾 

## Highlights
- [x] Good implementation 👍🏾 
- [x] Good PR description 💯 
- [x] Good commit history 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Navbar.js [hunk 2]
```diff
@@ -1,20 +1,34 @@
-import React from "react";
-import { Link } from "react-router-dom";
+import { Link } from 'react-router-dom';
+import React from 'react';
+// import './styles/Nav.css';
+// import login from '../images/login.png';
```
Comment (ahmetbozaci) [inline src/components/Navbar.js:4]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/Navbar.js [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
-        <Link to="Books" className="book-click">Books</Link>
-        <Link to="Categories" className="cat-click">Categories</Link>
+        <Link to="/categories" style={navStyle}>
+          <li>Categories</li>
+        </Link>
+      </ul>
+      <div className="login">
+        {/* <Link to="/" style={navStyle}>
+          <img src={login} alt="Book. logo" />
+        </Link> */}
```
Comment (ahmetbozaci) [inline src/components/Navbar.js:28]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/displayCategories.js [hunk 4]
```diff
@@ -1,19 +1,17 @@
 import React from 'react';
-import { useDispatch } from 'react-redux';
-import { checkBookStatus } from '../redux/Categories/Categories';
+import { useDispatch, useSelector } from 'react-redux';
+// import './styles/Categories.css';
```
Comment (ahmetbozaci) [inline src/components/displayCategories.js:3]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/eachBook.js [hunk 5]
```diff
@@ -1,31 +1,74 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+// import './styles/Book.css';
```
Comment (ahmetbozaci) [inline src/components/eachBook.js:1]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/eachBook.js [hunk 6]
```diff
@@ -1,31 +1,74 @@
...
 import { useDispatch } from 'react-redux';
-import { deleteBookFetch} from '../redux/Books/Books';
+// import { Progress } from 'antd';
```
Comment (ahmetbozaci) [inline src/components/eachBook.js:3]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/eachBook.js [hunk 7]
```diff
@@ -1,31 +1,74 @@
...
+              type="circle"
+              width={80}
+              format={() => ''}
+              strokeColor={{
+                '0%': '#307bbe',
+                '100%': '#379cf6',
+              }}
+              trailColor="#e8e8e8"
+              percent={book.completed}
+            /> */}
```
Comment (ahmetbozaci) [inline src/components/eachBook.js:54]: OPTIONAL
_I make it optional but this is a general requirement for every project._
- [ ] To make clean code kindly delete commented code.

File: src/components/inputForm.js [hunk 8]
```diff
@@ -1,36 +1,41 @@
...
+          <option value="Technology">Technology</option>
+          <option value="Fiction">Fiction</option>
+          <option value="Adventure">Adventure</option>
+          <option value="Action">Action</option>
+          <option value="Romance">Romance</option>
+        </select>
+        <button className="submit-btn" type="submit">Add Book</button>
+      </form>
+    </div>
+  );
```
Comment (ahmetbozaci) [inline src/components/inputForm.js:38]: - [ ] I can't see any form on the page. Kindly fix it.

image

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@Therealadetunji  ✋🏼 
#### Your project is excellent. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**